### PR TITLE
Tidy Projects: delete unused projects UI step definitions

### PR DESCRIPTION
**Pull request by Erin007**
We removed ability for students to publish projects to a public gallery for safety and customer support time management (see [spec](https://docs.google.com/document/d/1vU68tzXG72Z80MvM6xYEDwF5hLDf7V097P-0dDfpouc/edit)). When that work was done (https://github.com/code-dot-org/code-dot-org/pull/57473) a handful of UI tests related to the (un)publishing projects were updated. However, there were lingering unused step definitions. This PR removes the unused code. 🗑️ 

With the clutter gone, I noticed there were two very similar step definitions "I open the share dialog" and "I open the project share dialog". I consolidated them keeping the step definition that included a retry to avoid flakiness and the more descriptive name. Then I updated references to that step.  

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by ebeastlake — COMMENTED** [review]
Approved pending green Drone run! We can run the UI tests locally to make sure we don't have any surprises when this gets DTTed, right?

**Comment by Erin007** [comment]
> Approved pending green Drone run! We can run the UI tests locally to make sure we don't have any surprises when this gets DTTed, right?

good callout. I'll kick off a Drone run with the "test all browsers" tag before I merge to avoid surprises on test.

**Review by molly-moen — APPROVED** [review]

**Pull request merged**
